rs/p2p/artifact_downloader/src/fetch_stripped_artifact/stripper.rs

```diff
@@ -16,24 +28,100 @@ impl Strippable for ConsensusMessage {
     type Output = MaybeStrippedConsensusMessage;
 
     fn strip(self) -> Self::Output {
+        let unstripped_consensus_message_id = self.id();
+
         match self {
             // We only strip data blocks.
             ConsensusMessage::BlockProposal(block_proposal)
                 if block_proposal.as_ref().payload.payload_type()
                     == ic_types::consensus::PayloadType::Data =>
             {
-                MaybeStrippedConsensusMessage::StrippedBlockProposal(block_proposal.strip())
+                let mut proto = pb::BlockProposal::from(&block_proposal);
+
+                // Remove the ingress payload from the proto.
+                if let Some(block) = proto.value.as_mut() {
+                    block.ingress_payload = None;
+                }
+
+                let data_payload = block_proposal.content.as_ref().payload.as_ref().as_data();
+                // TODO(CON-1402): avoid the clone
+                let stripped_ingress_payload = data_payload.batch.ingress.clone().strip();
+
+                MaybeStrippedConsensusMessage::StrippedBlockProposal(StrippedBlockProposal {
+                    block_proposal_without_ingresses_proto: proto,
+                    stripped_ingress_payload,
+                    unstripped_consensus_message_id,
+                })
             }
             msg => MaybeStrippedConsensusMessage::Unstripped(msg),
         }
     }
 }
 
-impl Strippable for BlockProposal {
-    type Output = StrippedBlockProposal;
+impl Strippable for IngressPayload {
+    type Output = StrippedIngressPayload;
 
-    // TODO(kpop): implement this
     fn strip(self) -> Self::Output {
-        unimplemented!()
+        let ingresses: Vec<_> = self.try_into().expect(
+            "A valid ingress payload shouldn't fail when converting to a vector of ingresses",
+        );
+
+        let stripped_ingresses = ingresses
+            .into_iter()
+            .map(|ingress| {
+                let ingress_message_id = IngressMessageId::from(&ingress);
+
+                if ingress.count_bytes() > INGRESS_MESSAGE_SIZE_STRIPPING_THRESHOLD_BYTES {
+                    MaybeStrippedIngress::Stripped(ingress_message_id)
+                } else {
+                    MaybeStrippedIngress::Full(ingress_message_id, ingress)
+                }
```

Comment:
I think it is too early for this optimization. Especially since we will propagate the entire block in the worst case if all `msgs < INGRESS_MESSAGE_SIZE_STRIPPING_THRESHOLD_BYTES`.

Reply:
Sure, I've reverted this change